addon/-private/system/model/internal-model.js

```diff
@@ -885,17 +885,28 @@ export default class InternalModel {
     @method clearRelationships
     @private
   */
-  clearRelationships() {
+  clearRelationships(resetAll = false) {
     this.eachRelationship((name, relationship) => {
       if (this._relationships.has(name)) {
         let rel = this._relationships.get(name);
-        rel.clear();
-        rel.removeInverseRelationships();
+
+        if (resetAll === true) {
+          rel.clear();
+          rel.removeInverseRelationships();
```

Comment:
curious, do we actually want to remove the entire inverse here, or just remove our own internalModel form the inverse? If its just remove ourselves, then the `clear` flag feels less bad and the above recommendation to two methods can be disregarded.

Specifically:

```js
if (resetAll === true) {
  rel.clear();
}

rel.unloadInverseInternalModel(this);
```

Also, would the following name change improve clarity?

`rel.unloadInverseInternalModel(this);` -> `rel.unloadInternalModelFromInverse(this);`

--

This would seem to tidy up how we deal with inverses, spuriously destroying them must lead to other issues...